### DISCUSSION TITLE
How to manually trigger the q-img spinner?

### DISCUSSION THREAD

**Discussion by BenJackGill** [quasarframework/quasar#12828] [General - Components / Directives / etc]
When images are loading the `q-img` element shows a spinner to indicate it's loading status.

I would like to also programmatically trigger that spinner for other occassions, such as when an image is being uploaded to the back-end.

I checked the docs and as far as I understand it there is no way to achieve this.

Perhaps I'm wrong?

If this is not possible I please suggest you add a prop named something like `spinner` or `spinning` which can be set to `true` or `false` and triggers the spinner at will.

**Comment by metalsadman** [comment]
you can use the default slot https://codepen.io/metalsadman/pen/MWraMem?editors=1010

**Reply by BenJackGill** [reply]
Thanks that helped.

I eventually got the slot spinner to match the `q-img` spinner.

But it took a lot of tweaking to figure it out. So I'm still giving my vote for a simple `spinning` prop that taps into the default `q-img` spinner.

Here is my spinner slot code and matching css if anyone else is interested. But you should also read the code in link above to know how to apply it:

The slot code:
```vue
      <template #loading>
        <div class="loading-wrapper fit flex flex-center">
          <q-spinner-ios class="loading"></q-spinner-ios>
        </div>
      </template>
      <div
        v-if="props.isLoading"
        class="q-img__loading absolute-full flex flex-center"
      >
        <q-spinner-ios></q-spinner-ios>
      </div>
```
The css:
```css
.loading {
  color: #fff;
}
.loading-wrapper {
  background: rgba(0, 0, 0, 0.47);
}
```
